ketcher-autotests/tests/User-Interface/Editing-Tools/Copy-Cut-Paste/copy-cut-paste.spec.ts

```diff
@@ -285,6 +285,301 @@ test.describe('Copy/Cut/Paste Actions', () => {
+    await delay(DELAY_IN_SECONDS.TWO);
+    await selectAtomInToolbar(AtomButton.Nitrogen, page);
+    await clickOnAtom(page, 'C', 12);
+  });
+
+  test('Copy and paste the structure with attached data', async ({ page }) => {
+    /*
+    Test case: EPMLSOPKET-1727
+    Description: After the clicking the Copy button, the selected object not disappears.
+    After pasting three same structures located on canvas.
+    */
+    const x = 500;
+    const y = 300;
+    await openFileAndAddToCanvas('attached.mol', page);
+    await copyAndPaste(page);
+    await page.mouse.click(x, y);
+  });
+
+  test('Cut and Paste structure with attached data and edit', async ({
+    page,
+  }) => {
+    /*
+    Test case: EPMLSOPKET-1727
+    Description: The correct structure is pasted on the canvas.
+    All attached data are correctly rendered.
+    User is able to edit the pasted structure.
+    */
+    await openFileAndAddToCanvas('attached.mol', page);
+    await cutAndPaste(page);
+    await page.mouse.click(CANVAS_CLICK_X, CANVAS_CLICK_Y);
+    await delay(DELAY_IN_SECONDS.TWO);
+    await selectAtomInToolbar(AtomButton.Nitrogen, page);
+    await clickOnAtom(page, 'C', 12);
+  });
+
+  test('Copy and paste Stereo structure with Chiral flag', async ({ page }) => {
+    /*
+    Test case: EPMLSOPKET-1728
+    Description: Copied objects are pasted correctly.
+    The structure is copied (and then is pasted) with the Chiral flag.
+    */
+    const x = 500;
+    const y = 300;
+    await openFileAndAddToCanvas('chiral-structure.mol', page);
+    await copyAndPaste(page);
+    await page.mouse.click(x, y);
+  });
+
+  test('Cut and Paste structure with Stereo and Chiral flag and edit', async ({
+    page,
+  }) => {
+    /*
+    Test case: EPMLSOPKET-1728
+    Description: Copied objects are pasted correctly.
+    The structure is cut (and then is pasted) with the Chiral flag.
+    */
+    await openFileAndAddToCanvas('chiral-structure.mol', page);
+    await cutAndPaste(page);
+    await page.mouse.click(CANVAS_CLICK_X, CANVAS_CLICK_Y);
+    await delay(DELAY_IN_SECONDS.TWO);
+    await selectAtomInToolbar(AtomButton.Nitrogen, page);
+    await clickOnAtom(page, 'C', 12);
+  });
+
+  test('Copy and paste reaction by hotkeys', async ({ page }) => {
+    /*
+    Test case: EPMLSOPKET-1730
+    Description: Copied objects are pasted correctly.
+    The structure is copied (and then is pasted) with reaction arrow and plus sign
+    */
+    const x = 500;
+    const y = 200;
+    await openFileAndAddToCanvas('reaction.rxn', page);
+    await copyAndPaste(page);
+    await page.mouse.click(x, y);
+  });
+
+  test('Cut and Paste reaction by hotkeys and edit', async ({ page }) => {
+    /*
+    Test case: EPMLSOPKET-1730
+    Description: Copied objects are pasted correctly.
+    The structure is cut (and then is pasted) with reaction arrow and plus sign
+    */
+    await openFileAndAddToCanvas('reaction.rxn', page);
```

Comment:
May be it is worth creating a separate function? 
You're repeating all the tests, but the only structure changes